src/node.cc

```diff
@@ -2579,18 +2579,38 @@ void Init(std::vector<std::string>* argv,
     // [0] is expected to be the program name, fill it in from the real argv.
     env_argv.push_back(argv->at(0));
 
-    // Split NODE_OPTIONS at each ' ' character.
-    std::string::size_type index = std::string::npos;
-    do {
-      std::string::size_type prev_index = index;
-      index = node_options.find(' ', index + 1);
-      if (index - prev_index == 1) continue;
-
-      const std::string option = node_options.substr(
-          prev_index + 1, index - prev_index - 1);
-      if (!option.empty())
-        env_argv.emplace_back(std::move(option));
-    } while (index != std::string::npos);
+    bool will_start_new_arg = true;
+    for (std::string::size_type index = 0;
+         index < node_options.size();
+         ++index) {
+      char c = node_options.at(index);
+      bool is_sep = false;
+
+      // Backslashes escape the following character
+      if (c == '\\') {
+        if (index + 1 == node_options.size()) {
+          fprintf(stderr,
+                  "%s: invalid escaping for NODE_OPTIONS\n",
+                  argv->at(0).c_str());
+          exit(9);
+        } else {
+          c = node_options.at(++index);
+        }
+      } else if (c == ' ') {
+        is_sep = true;
+      }
+
+      if (is_sep) {
+        will_start_new_arg = true;
+      } else {
+        if (will_start_new_arg) {
+          env_argv.push_back(std::string(1, c));
```

Comment:
From a raw perf perspective the string created here could benefit from a `reserve` call to a specific value (like 10 or so, like most Node options), but I figured the added complexity (a magic number) wasn't worth the gain (which would likely be extremely negligible in practice).

Reply:
Yeah, I think this should be okay for our purposes here :)